services/web-app/components/markdown/image.js

```diff
@@ -27,10 +27,8 @@ const getImageProps = (props) => {
   }
 
   // do not optimize if is svg file (absolute or relative)
-  if (
-    src.endsWith('.svg') ||
-    (isAbsoluteUrl(src) && new URL(src).pathname.split('/').pop().endsWith('.svg'))
-  ) {
+  const url = isAbsoluteUrl(src) ? new URL(src).pathname.split('/').pop().endsWith('.svg') : null
+  if (src.endsWith('.svg') || !!url) {
```

Comment:
For the cases in this PR where `isAbsoluteUrl` is being used and, if true, another URL object is created, it's actually doing more harm than good. Rather than double-instantiating the same URL, this is one of those cases where it's actually a better idea to inline this logic as opposed to using a utility function.

Right now, the logic basically does:
`if new URL(src) doesn't throw and new URL(src).pathname.split....., then ....`

Instead of this, an alternative solution would be to put the instantiation/try/catch right here in this function

```ts
let url
try {
  url = new URL(src)
} catch {}

if (src.endsWith('.svg') || url?.pathname.endsWith('.svg')) {
  ...
}
```

or, the utility function could return the instantiated object as opposed to throwing it away:
```ts
// previously isAbsoluteUrl()
function createUrl(src) {
  let result: URL | undefined
  try {
    result = new URL(src)
  } catch {}

  return result
}

// (later)
const url = createUrl(src)

if (url && someOtherCondition) {
  ...
}
```

Reply:
done!